sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -153,7 +153,12 @@ object FunctionRegistryBase {
         } catch {
           // the exception is an invocation exception. To get a meaningful message, we need the
           // cause.
-          case e: Exception => throw new AnalysisException(e.getCause.getMessage)
+          case e: InvocationTargetException =>
+            e.getCause() match {
+              // Propagate Error Class AnalysisExceptions
+              case a: AnalysisException if (a.getErrorClass() != null) => throw a
```

Comment:
I would propose to propagate any SparkThrowable otherwise wrap by AnalysisException like in https://github.com/apache/spark/pull/35467